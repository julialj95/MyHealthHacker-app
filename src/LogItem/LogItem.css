.log_record {
  display: flex;
  border: solid 1px black;
  padding: 5px 0 5px 0;
  margin: 5px;
  background-color: #6e3667;
  border: solid 5px #88d317;
  border-radius: 10px;
  font-family: "Nunito";
  color: white;
}

.short {
  flex-direction: row;
  justify-content: space-between;
}
.long {
  flex-direction: column;
}

.title_row {
  flex: 1;
}

.top_section {
  display: flex;
  flex-direction: column;
}

.log_section {
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  border: solid 3px #88d317;
  border-radius: 5px;
}

.button_section {
  display: flex;
  flex-direction: column;
}

.logItem_btn {
  flex: 1;
  margin: 5px 10px 5px 10px;
  background-color: #3c3c3c;
  box-shadow: 1px 1px white;
  border: 2px solid #88d317;
  border-radius: 3px;
  color: white;
  font-family: "Nunito";
  cursor: pointer;
}

.date_display {
  font-size: 18px;
  margin-left: 5px;
}
.view_btn {
  margin: 5px 10px 5px 10px;
  background-color: #3c3c3c;
  box-shadow: 1px 1px white;
  border: 2px solid #88d317;
  border-radius: 3px;
  color: white;
  font-family: "Nunito";
  font-size: 14px;
  padding: 5px;
  cursor: pointer;
}

.tracker {
  font-size: 16px;
  flex-grow: 1;
  color: white;
  padding: 10px;
  border-radius: 5px;
}
.margin {
  margin-top: 23px;
  margin-bottom: 23px;
}
input[type="range"] {
  -webkit-appearance: none;
  /* width: 100%; */
}

.sleep_style::-webkit-slider-runnable-track,
.mood_style::-webkit-slider-runnable-track {
  width: 70%;
  height: 5px;
  background: linear-gradient(45deg, red, yellow, green);
  border: none;
  border-radius: 3px;
}

.stress_style::-webkit-slider-runnable-track {
  width: 70%;
  height: 5px;
  background: linear-gradient(45deg, green, yellow, red);
  border: none;
  border-radius: 3px;
}

input[type="range"]::-webkit-slider-thumb {
  -webkit-appearance: none;
  border: none;
  height: 16px;
  width: 16px;
  border-radius: 50%;
  background: #88d317;
  margin-top: -4px;
}

.edit_log_form {
  text-align: center;
  background-color: #1a0135;
  /* margin-left: 10%;
  margin-right: 10%; */
  border: solid 5px red;
  border-radius: 15px;
  color: white;
}
@media all and (min-width: 600px) {
  .button_section {
    flex-direction: row;
    justify-content: space-between;
  }

  .date_display {
    font-size: 21px;
  }
  .view_btn {
    font-size: 18px;
  }
  .logItem_btn {
    font-size: 18px;
  }
  .tracker {
    font-size: 19px;
  }
}
@media all and (min-width: 760px) {
  .top_section {
    flex-direction: row;
  }
}
@media all and (min-width: 960px) {
  .date_display {
    font-size: 26px;
  }
  .view_btn {
    font-size: 20px;
  }
  .logItem_btn {
    font-size: 20px;
  }
  .tracker {
    font-size: 23px;
  }
}
