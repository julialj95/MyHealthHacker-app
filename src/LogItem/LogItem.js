import React from "react";
import "./LogItem.css";

class LogItem extends React.Component {
  constructor() {
    super();
    this.state = {
      displayFull: false,
    };
  }
  renderFullItem() {
    return (
      <section className="log_record long">
        <h2 className="title_row">{this.props.date}</h2>
        <div className="top_section">
          <div className="log_section">
            <p>Stress Level: {this.props.stress}/5</p>
            <p>Mood: {this.props.mood}/5</p>
          </div>
          <div className="log_section">
            <p>Hours slept: {this.props.sleep_hours}</p>
            <p>Sleep Quality: {this.props.sleep_quality}/5</p>
          </div>
          <div className="log_section">
            <p>Minutes of Exercise: {this.props.exercise}</p>
            <p>Type of Exercise: {this.props.exercise_type}</p>
            <p>Ounces of water consumed: {this.props.water}</p>
          </div>
        </div>
        <p>{this.props.notes}</p>
        <button onClick={this.editLog}>Edit</button>
        <button>Delete</button>
        <button onClick={this.setState({ displayFull: false })}>
          Collapse Log
        </button>
      </section>
    );
  }
  renderPartialItem() {
    return (
      <section className="log_record short">
        <h2>{this.props.date}</h2>
        <button onClick={this.setState({ displayFull: true })}>View Log</button>
      </section>
    );
  }
  render() {
    return (
      <div>
        {this.state.displayFull
          ? this.renderFullItem()
          : this.renderPartialItem()}
      </div>
    );
  }
}
export default LogItem;
