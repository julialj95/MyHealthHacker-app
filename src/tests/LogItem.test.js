import React from "react";
import ReactDOM from "react-dom";
import LogItem from "../LogItem/LogItem";
import { MemoryRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(
    <Router>
      <LogItem />
    </Router>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <Router>
        <LogItem />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
