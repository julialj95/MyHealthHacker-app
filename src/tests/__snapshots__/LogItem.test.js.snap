// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <section
    className="log_record short"
  >
    <h2>
      2021-04-30
    </h2>
    <button
      onClick={[Function]}
    >
      View Log
    </button>
  </section>
</div>
`;
